### DISCUSSION TITLE
Self hosted auth.signIn with an auth provider leads to "No API key found in request" from Kong

### DISCUSSION THREAD

**Discussion by prulowuli** [supabase/supabase#2117] [Questions] [answered]
I've set up a self hosted version of Supabase locally (on localhost:8000) using the docker compose provided in the repo and added the relevant gotrue environment variables to docker-compose.yml. Frontend is written in Next.js.

Using the Auth component from https://github.com/supabase/ui which uses supabaseClient.auth.signIn internally, I am routed to `http://localhost:8000/auth/v1/authorize?provider=google` (NEXT_PUBLIC_SUPABASE_URL = http://localhost:8000) after clicking on the "Sign up with google" button which then shows `Kong Error No API key found in request`. 

I believe the navigation to the `/auth/v1/authorize` path does not include the `NEXT_PUBLIC_SUPABASE_ANON_KEY` in the header. 

Did I miss out a step during configuration of the self hosted Supabase or should the anon key not be required for that path? I don't have issues with the email+password signup.

**Comment by awalias** [comment, marked as answer]
hey @prulowuli ! Nice catch, we were indeed missing some kong bypass routes in the kong.yml inside the CLI, I've now fixed this here: https://github.com/supabase/cli/commit/cd8d4cb9bd707c6ab25c395b9f7f3872ce951a2c

**Reply by awalias** [reply]
you can run `docker compose --file .supabase/docker/docker-compose.yml --project-name supabase up --build --detach` to rebuild the local containers once you have pulled the new version

**Reply by prulowuli** [reply]
Hi @awalias, thanks for the quick response! I'm actually using the docker-compose found in `supabase/supabase`'s docker folder which seems to contain a different kong.yml from the one in `supabase/cli` even before your latest commit. I was using the 2nd option shown here: https://supabase.io/docs/guides/self-hosting

Is the cli the best way to start a self hosted project or is cloning the main repo a viable option too?
